packages/dds/tree/src/test/shared-tree/summaryBenchmarkSizes.ts

```diff
@@ -0,0 +1,221 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import { IsoBuffer } from "@fluidframework/common-utils";
+import { makeRandom } from "@fluid-internal/stochastic-test-utils";
+import { TransactionResult } from "../../checkout";
+import { FieldKinds, PlacePath, singleTextCursor } from "../../feature-libraries";
+import { ISharedTree } from "../../shared-tree";
+import { detachedFieldAsKey, FieldKey, rootFieldKey, TreeValue } from "../../tree";
+import { brand } from "../../util";
+import { TestTreeProvider } from "../utils";
+import { fieldSchema, GlobalFieldKey, namedTreeSchema, SchemaData } from "../../schema-stored";
+
+const globalFieldKey: GlobalFieldKey = brand("globalFieldKey");
+
+describe("Summary size benchmark", () => {
+    it("for different sized trees", async () => {
+        const output = await getSummaryBenchmarkSizes();
+        assert(output.summarySizeWideInsert1 < 1000);
+        assert(output.summarySizeWideInsert10 < 3000);
+        assert(output.summarySizeWideInsert100 < 20000);
+        assert(output.summarySizeNarrowInsert10 < 3000);
+        assert(output.summarySizeNarrowInsert100 < 20000);
+        assert(output.summarySizeNoEdits < 1000);
+    });
+    it("for 1000 inserts width wise", async () => {
+        await assert.rejects(getInsertsSummarySize(1000, 0, true), { message: "BatchTooLarge" });
+    });
+});
+
+export interface summarySizeBenchmarks {
```

Comment:
I see no benefit to having this type.

Wouldn't it be simpler to inline the calls to getInsertsSummarySize in the test above?

Remember that the point of using abstraction is usually to make things easier to change or easier to reason about.
In this code, if we wanted to add more test case, or change the size of one test case, we have to change this type, adding a member, giveing it the special name and comment, add a line to the above test, add a line to initialize it to 0, and add a line to set it to its actual value. Forget that last step and the test will silently pass but not actually check anything.

Compare this to just calling the getInsertsSummarySize inline in the test above. Adding a new test case or updating an existing one is a one line change, and there is no way to accidently forget to set the size. Thats way simpler to maintain, and less code to think about.

Another approach is table driven testing where you define a list of test cases and loop over them. I don't think thats needed here, but it would be cleaner than what you have and somewhat similar in style.

Reply:
Removed interface, and separated test cases